File: 백준/Bronze/5575. 타임 카드/타임 카드.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Problem struct {
	*bufio.Reader
	*bufio.Writer
	*StringBuilder
	*Data
}

type Data struct {
	a Pair
	b Pair
	c Pair
}

type Pair struct {
	begin    Work
	end      Work
	duration int
}

type Work struct {
	hour   int
	minute int
	second int
	stamp  int
}

func NewProblem() *Problem {
	return &Problem{
		Reader:        bufio.NewReader(os.Stdin),
		Writer:        bufio.NewWriter(os.Stdout),
		StringBuilder: NewStringBuilder(),
		Data:          NewData(),
	}
}

func NewData() *Data {
	return &Data{}
}

func (p *Problem) Solve() {
	p.input()
	p.logic()
	p.output()
}

func (p *Problem) input() {
	p.a.input(p)
	p.b.input(p)
	p.c.input(p)
}

func (p *Pair) input(problem *Problem) {
	p.begin.input(problem)
	p.end.input(problem)
	p.duration = p.end.stamp - p.begin.stamp
}

func (w *Work) input(problem *Problem) {
	if _, err := fmt.Fscan(problem.Reader, &w.hour, &w.minute, &w.second); err != nil {
		os.Exit(1)
	}
	w.stamp = w.hour*60*60 + w.minute*60 + w.second
}

func (p *Problem) logic() {
	p.formatAppend(durationToTime(p.a.duration))
	p.formatAppend(durationToTime(p.b.duration))
	p.formatAppend(durationToTime(p.c.duration))
}

func (p *Problem) formatAppend(hour, minute, second int) {
	p.Append(hour).Append(" ").Append(minute).Append(" ").Append(second).Append("\n")
}

func durationToTime(duration int) (int, int, int) {
	second := duration % 60
	minute := duration / 60
	hour := minute / 60
	minute %= 60
	return hour, minute, second
}

func (p *Problem) output() {
	_, err := fmt.Fprint(p.Writer, p.String())
	if err != nil {
		os.Exit(1)
	}
}

func main() {
	p := NewProblem()
	defer func(p *Problem) {
		_ = p.Flush()
	}(p)
	p.Solve()
}

type StringBuilder struct {
	strings.Builder
}

func NewStringBuilder() *StringBuilder {
	var sb StringBuilder
	sb.Grow(30)
	return &sb
}

func (sb *StringBuilder) Append(values ...interface{}) *StringBuilder {
	for _, value := range values {
		sb.WriteString(convert(value))
	}
	return sb
}

func convert(value interface{}) string {
	switch v := value.(type) {
	case string:
		return v
	case int:
		return strconv.Itoa(v)
	case int8:
		return strconv.Itoa(int(v))
	case int16:
		return strconv.Itoa(int(v))
	case int32:
		return strconv.Itoa(int(v))
	case int64:
		return strconv.FormatInt(v, 10)
	case uint:
		return strconv.FormatUint(uint64(v), 10)
	case uint8:
		return strconv.FormatUint(uint64(v), 10)
	case uint16:
		return strconv.FormatUint(uint64(v), 10)
	case uint32:
		return strconv.FormatUint(uint64(v), 10)
	case uint64:
		return strconv.FormatUint(v, 10)
	case float32:
		return strconv.FormatFloat(float64(v), 'f', -1, 32)
	case float64:
		return strconv.FormatFloat(v, 'f', -1, 64)
	default:
		return fmt.Sprintf("%v", v)
	}
}

File: 백준/Bronze/5575. 타임 카드/타임 카드.rs
use std::io;

struct Problem {
    vec: Vec<Data>,
    answer: String,
}

struct Data {
    duration: u32,
}

struct Work {
    stamp: u32,
}


impl Problem {
    fn new() -> Problem {
        Problem {
            vec: Vec::new(),
            answer: String::new(),
        }
    }

    fn input(&mut self) {
        for _i in 0..3 {
            let mut input = String::new();
            io::stdin().read_line(&mut input).unwrap();
            let mut input = input.split_ascii_whitespace().map(|e| e.parse::<u32>().unwrap());
            let mut input = || input.next().unwrap();
            self.vec.push(Data::new(Work::new(input(), input(), input()), Work::new(input(), input(), input())));
        }
    }

    fn solve(&mut self) {
        self.format_push(self.vec.get(0).unwrap().duration_to_time());
        self.format_push(self.vec.get(1).unwrap().duration_to_time());
        self.format_push(self.vec.get(2).unwrap().duration_to_time());
    }

    fn format_push(&mut self, time: (u32, u32, u32)) {
        self.answer.push_str(&time.0.to_string());
        self.answer.push(' ');
        self.answer.push_str(&time.1.to_string());
        self.answer.push(' ');
        self.answer.push_str(&time.2.to_string());
        self.answer.push('\n');
    }

    fn output(&self) {
        println!("{}", self.answer);
    }
}

impl Data {
    fn new(begin: Work, end: Work) -> Data {
        Data {
            duration: &end.stamp - &begin.stamp,
        }
    }

    fn duration_to_time(&self) -> (u32, u32, u32) {
        let second = self.duration % 60;
        let mut minute = self.duration / 60;
        let hour = minute / 60;
        minute %= 60;
        (hour, minute, second)
    }
}

impl Work {
    fn new(hour: u32, minute: u32, second: u32) -> Work {
        Work {
            stamp: hour * 60 * 60 + minute * 60 + second,
        }
    }
}

fn main() {
    let mut problem = Problem::new();
    problem.input();
    problem.solve();
    problem.output();
}